sdk_container/src/third_party/coreos-overlay/sys-kernel/coreos-sources/files/6.6/z0004-efi-Add-an-EFI_SECURE_BOOT-flag-to-indicate-secure-b.patch

```diff
@@ -0,0 +1,166 @@
+From 1e2ffbec195c89d887bc088691ebb19c9173ecad Mon Sep 17 00:00:00 2001
```

Comment:
These patches will be a pain to maintain, given that there is an agressive backport policy for the stable branches happening lately. But I do not see a way out either, other than trying to push for / advocating to have these patches merged in Linux master.

For example, the small pahole patch that we use has to updated at least a few times a year, these core patches probably more often and with way more care.

There are also great security implications for these patches and my suggestion is to have them peer reviewed from the broader Linux community by sending them upstream first.

Reply:
These patches have been posted and rejected upstream by the kernel community. Still, they are required to participate in the community shim signing process and all distros carry them. I agree that it they will occasionally break and require us to manually fix them up.

Reply:
My main concern is the following - once a week, there is a new patch level release for the linux kernel versions that we use (currently just 1 now - 6.6, might be in the future 6.12 or even faster 6.10) - There is a chance that the patches for secureboot have conflicts and inevitably, this will lead to two things: fast-forward conflict resolution or less often than once a week kernel upgrades. The first part is the most important - because these patches affect core kernel pathways, the changes need to be carefully crafted and reviewed so that there is less chance of a critical vulnerability.

The discussion I would like to start is - this feature is going to be used right away on all the clouds? or just on a subset and if there is just a subset, we can just silo it to that subset and in the worst case scenario, there is less chance to have something affecting all the images.

tl;dr if these patches are not properly reviewed and maintained, there is a chance we have less security instead of more.